test/async-script-loader-spec.js

```diff
@@ -14,87 +14,146 @@ class MockedComponent extends React.Component {
+    let scriptLoaded = false;
+    const asyncScriptOnLoadSpy = (entry) => {
+      scriptLoaded = entry.loaded;
+      scriptErrored = entry.errored;
+      asyncScriptOnLoadCalled = true;
+    }
+    const ComponentWrapper = makeAsyncScriptLoader(URL)(MockedComponent);
     assert.equal(ComponentWrapper.displayName, "AsyncScriptLoader(MockedComponent)");
     const instance = ReactTestUtils.renderIntoDocument(
-      <ComponentWrapper />
+      <ComponentWrapper asyncScriptOnLoad={asyncScriptOnLoadSpy} />
     );
+    documentLoadScript(URL);
+
     assert.ok(ReactTestUtils.isCompositeComponent(instance));
     assert.ok(ReactTestUtils.isCompositeComponentWithType(instance, ComponentWrapper));
     assert.isNotNull(ReactTestUtils.findRenderedComponentWithType(instance, MockedComponent));
-    assert.equal(hasScript(URL), true);
+    assert.equal(hasScript(URL), true, "Url in document");
+    assert.equal(asyncScriptOnLoadCalled, true, "asyncScriptOnLoad callback called");
+    assert.equal(scriptLoaded, true, "script loaded state set");
+    assert.equal(scriptErrored, undefined, "script errored state unset");
+  });
+
+  it("should fire asyncScriptOnLoad on errored script load", () => {
+    const URL = "http://example.com/?errored=true";
+    let asyncScriptOnLoadCalled = false;
+    let scriptErrored = false;
+    let scriptLoaded = false;
+    const asyncScriptOnLoadSpy = (entry) => {
+      scriptErrored = entry.errored;
+      scriptLoaded = entry.loaded;
+      asyncScriptOnLoadCalled = true;
+    }
+    const ComponentWrapper = makeAsyncScriptLoader(URL)(MockedComponent);
+    ReactTestUtils.renderIntoDocument(
+      <ComponentWrapper asyncScriptOnLoad={asyncScriptOnLoadSpy} />
+    );
+    documentErrorScript(URL);
+
+    assert.equal(hasScript(URL), true, "Url in document");
+    assert.equal(asyncScriptOnLoadCalled, true, "asyncScriptOnLoad callback called");
+    assert.equal(scriptErrored, true, "script errored state set");
+    assert.equal(scriptLoaded, false, "script loaded state set");
   });
-  it("should handle successfully already loaded global object", () => {
-    const URL = "http://example.com";
+
+  it("should handle successfully already loaded global object and fire asyncScriptOnLoad", () => {
+    const URL = "http://example.com/?global=true";
     const globalName = "SomeGlobal";
     window[globalName] = {};
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, URL, { globalName: globalName });
+    let asyncScriptOnLoadCalled = false;
+    const asyncScriptOnLoadSpy = () => {
+      asyncScriptOnLoadCalled = true;
+    }
+    const ComponentWrapper = makeAsyncScriptLoader(URL, { globalName: globalName })(MockedComponent);
     const instance = ReactTestUtils.renderIntoDocument(
-      <ComponentWrapper />
+      <ComponentWrapper asyncScriptOnLoad={asyncScriptOnLoadSpy} />
     );
-    assert.equal(hasScript(URL), true);
+
+    assert.equal(hasScript(URL), false, "Url not in document");
```

Comment:
this one was "wrong" before... it was checking the previous tests url instead of verifying that this tests url was **not** in the document